api/client.go

```diff
@@ -661,6 +663,25 @@ func NewClient(c *Config) (*Client, error) {
 		client.setNamespace(namespace)
 	}
 
+	if envHeaders := os.Getenv(EnvVaultHeaders); envHeaders != "" {
+		var result map[string]any
+		err := json.Unmarshal([]byte(envHeaders), &result)
```

Comment:
Parsing JSON objects is the most easily-understood way to support > 1 header specified in the same environment variable. However, it limits each key to a single value (which is not a restriction of HTTP). Is this sufficient, or should an alternative be considered?

Reply:
To me, this seems reasonable. It's imperfect, but it follows the same pattern as is done with the `-header` option, as far as I can tell: https://github.com/hashicorp/vault/pull/12508